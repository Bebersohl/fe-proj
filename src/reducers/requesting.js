const requesting = (state = false, action) => {
  switch (action.type) {
    case 'SIGN_IN_USER_REQUEST':
      return true
    case 'SIGN_IN_USER_SUCCESS':
    case 'SIGN_IN_USER_FAIL':
      return false
    default:
      return state
  }
}

export default requesting