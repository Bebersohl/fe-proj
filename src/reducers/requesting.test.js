import requesting from './requesting'

it('handles initial state', () => {
  const next = requesting(undefined, {})
  expect(next).toMatchSnapshot()
})

it('handles SIGN_IN_USER_REQUEST', () => {
  const next = requesting(undefined, {type: 'SIGN_IN_USER_SUCCESS'})
  expect(next).toMatchSnapshot()
})

it('handles SIGN_IN_USER_SUCCESS', () => {
  const next = requesting(undefined, {type: 'SIGN_IN_USER_SUCCESS'})
  expect(next).toMatchSnapshot()
})

it('handles SIGN_IN_USER_FAIL', () => {
  const next = requesting(undefined, {type: 'SIGN_IN_USER_FAIL'})
  expect(next).toMatchSnapshot()
})
