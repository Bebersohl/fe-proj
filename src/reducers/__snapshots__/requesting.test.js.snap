exports[`test handles SIGN_IN_USER_FAIL 1`] = `false`;

exports[`test handles SIGN_IN_USER_REQUEST 1`] = `false`;

exports[`test handles SIGN_IN_USER_SUCCESS 1`] = `false`;

exports[`test handles initial state 1`] = `false`;
